test/functional/api/v2/fact_value_extensions_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+module Api
+  module V2
+    class FactValuesControllerExtensionsTest < ActionController::TestCase
+      setup do
+        User.current = User.find_by_login "admin"
+        @facts       = {
+            "interfaces"        => "lo,eth0",
+            "ipaddress"         => "192.168.100.42",
+            "ipaddress_eth0"    => "192.168.100.42",
+            "macaddress_eth0"   => "AA:BB:CC:DD:EE:FF",
+            "discovery_bootif"  => "AA:BB:CC:DD:EE:FF",
+            "memorysize_mb"     => "42000.42",
+            "discovery_version" => "3.0.0",
+        }
+        FactoryGirl.create(:setting,
```

Comment:
Since I'm using `import_host_and_facts` it needs the setting for hostname or it will return the error:
```
Foreman::Exception: ERF42-9632 [Foreman::Exception]: Invalid facts: hash does not contain a valid value for any of the facts in the discovery_hostname setting:
```
I'll remove the prefix setting but hostname needs to stay.